tests/test_cssselect.py

```diff
@@ -505,6 +511,14 @@ def xpath(css):
             "e[not(count(preceding-sibling::*) mod 2 = 0)]")
         assert xpath('e:nOT(*)') == (
             "e[0]")  # never matches
+        assert xpath("e:has(> f)") == "e[./f]"
+        assert xpath("e:has(f)") == "e[descendant::f]"
+        assert xpath("e:has(~ f)") == "e[following-sibling::f]"
+        assert (
+            xpath("e:has(+ f)")
```

Comment:
I think this can be simplified to `e[following-sibling::*[(name() = 'f') and (position() = 1)]]` as well, or am I wrong (or is it hard to implement)?